tuf/testutils/repo.go

```diff
@@ -36,7 +36,13 @@ func CreateKey(cs signed.CryptoService, gun, role string) (data.PublicKey, error
 		if err != nil {
 			return nil, err
 		}
-		key = data.NewECDSAx509PublicKey(trustmanager.CertToPEM(cert))
+		// Keep the x509 key type consistent with the key's algorithm
+		if keyAlgorithm == data.RSAKey {
+			key = data.NewRSAx509PublicKey(trustmanager.CertToPEM(cert))
+		} else {
```

Comment:
I think it's technically impossible since the call to `cs.Create()` will reject any different type, but agreed that it would be nicer.